main.js

```diff
@@ -26,17 +26,16 @@ function createProduct(products) {
   $productName.innerHTML = products.name + '<br> $' + products.price;
   $icons.append($heartIcon, ' ', $cartPlus);
 
-  $heartIcon.addEventListener('click', function() {
-    if ($heartIcon.className !== 'fa fa-heart fav') {
-      data.favorites.push(products);
-      $heartIcon.className += ' fav';
-    }
-  });
-
-  $cartPlus.addEventListener('click', function() {
-    if ($cartPlus.className !== 'fa fa-cart-plus in-cart') {
-      data.cart.push(products);
-      $cartPlus.className += ' in-cart';
+  $product.addEventListener('click', function(event) {
+    if (event.target.tagName.toLowerCase() === 'i') {
+      if (event.target.className === 'fa fa-heart') {
+        data.favorites.push(products);
+        $heartIcon.className += ' fav';
+      }
+      if (event.target.className === 'fa fa-cart-plus') {
+        data.cart.push(products);
+        $cartPlus.className += ' in-cart';
+      }
```

Comment:
This adds an event listener on each product that is created. Add it in on the product container itself.

Only difference is that you have to iterate through the product list to find the product with the same product id.

You can grab the product id from the card using the closest method and the dataset property

https://developer.mozilla.org/en-US/docs/Web/API/Element/closest
https://developer.mozilla.org/en-US/docs/Web/API/HTMLOrForeignElement/dataset